### DISCUSSION TITLE
Checkout (Fire checkout module) require.js issues

### DISCUSSION THREAD

**Discussion by EligijusGricius** [breezefront/community#92] [1. Help] [answered]
Hi, I have some strange issues with checkout (fire checkout installed), sometimes while it is loading I get a conflict with require.js. Require function needed to include jquery (require(['jquery'], function($){}). Error message:  "Uncaught Error: Mismatched anonymous define() module: function () {".  What can it be? Thanks!

https://prnt.sc/PLS25qHW-NSD

**Comment by vovayatsyuk** [comment]
I don't know what's causing this error. 

Is it a pure Magento and Breeze or have you added your custom scripts? Maybe you are using custom js in firecheckout? https://docs.swissuplabs.com/m2/extensions/firecheckout/configuration/#custom-css-and-js-settings-section

p.s. JFYI, Breeze is automatically disabled at checkout and cart pages, so this issue is not related to Breeze.

**Comment by EligijusGricius** [comment]
Hi, it has some custom scripts. I am not using any custom.js (Swissup_Firecheckout/web/js/custom.js) related to the plugin, I use only global theme custom.js (theme/web/js/custom.js) it has an inside `require` function which includes `jquery` support. I did some testing and the issue comes 2/3 of 10 reloads. I am working on VMware, can it cause a problem?

**Reply by vovayatsyuk** [reply]
Can you show the contents of your `theme/web/js/custom.js`?

**Reply by EligijusGricius** [reply]
Yea sure, added to my server, because git editor breaks down the code: [custom.js](https://daddyisdev.com/custom.js)

**Reply by vovayatsyuk** [reply]
You don't need a `require` inside the js file. `require()` is for inline scripts only.

You need to use define:

```js
define(['jquery', 'domReady!'], function ($) {
    //
});
```

**Reply by EligijusGricius** [reply]
Thanks, but it didn't helped, now I get another issue: `Uncaught ReferenceError: define is not defined`, how can I fix it?

**Reply by vovayatsyuk** [reply]
Can you share the URL to your site? Checkout page is not powered by Breeze, so it's an issue with Magento scripts and I don't know what is wrong here.

**Reply by EligijusGricius** [reply]
I can't share it because it's on virtual machine. Thanks for help for now. :)

**Reply by EligijusGricius** [reply]
Hi, we still facing this issue, can you check it on live page? http://dssriga.wdmarket.technology/ Right now I just made a workaround: 
`let currentUrl = window.location.href;
if(!currentUrl.includes("checkout")) {`

and still using  `require(["jquery"], function($){ define(["domReady!"], function () {` otherthise jquery doesn't wok

Thanks!

**Reply by vovayatsyuk** [reply, marked as answer]
You are trying to use the script across all pages, but that will not work well, because checkout and cart pages are not breeze-js based. You have to include your custom script at breeze-js pages only.

1. Follow this instruction and your js file will be added only at breeze.js pages: https://breezefront.com/docs/custom-javascript
2. If also need a custom.js for the checkout page, use the firecheckout custom.js feature: http://docs.swissuplabs.com/m2/extensions/firecheckout/customization/custom-js/

**Reply by EligijusGricius** [reply]
Thanks Vova, custom.js worked
